quinn-proto/src/connection/mod.rs

```diff
@@ -2183,14 +2189,11 @@ impl Connection {
     /// state of these subsystems is no longer valid or optimal. In this case it might be
     /// faster or reduce loss to settle on optimal values by restarting from the initial
     /// configuration in the [`TransportConfig`].
-    pub fn path_changed(&mut self, now: Instant) {
-        // TODO(@divma): evaluate how this is used
-        // wrong call in the multipath case anyhow
-        self.paths
-            .get_mut(&PathId::ZERO)
-            .expect("this might fail")
-            .data
-            .reset(now, &self.config);
+    pub fn path_changed(&mut self, path_id: PathId, now: Instant) -> bool {
+        let config = self.config.clone(); // :/
+        self.path_mut(path_id)
+            .map(|d| d.reset(now, &config))
+            .is_some()
```

Comment:
we might just not need this whole thing anymore now we're on multipath? i guess we could keep it though.